misk/src/main/kotlin/misk/environment/Environment.kt

```diff
@@ -12,13 +13,27 @@ enum class Environment {
   companion object {
     internal val logger = getLogger<Environment>()
     private const val ENV_ENVIRONMENT = "ENVIRONMENT"
+    private lateinit var env: Environment
+
+    fun setTesting() {
+      if (::env.isInitialized) {
```

Comment:
is this a safeguard in case this is called out of band?

Reply:
it's to allow the env var to take precedence. I think eventually we'd want to this to be a set once property with no default. until then having it prefer the env var is probably the safest rollout

Reply:
ah because we have scripts already passing in ENVIRONMENT=TESTING, gotcha

Reply:
yea. I think we'll still need that env var set, because we don't enforce _all_ tests use the `MiskTestExtension`. although those tests are probably not reading the env var anyways, since they don't use guice. so probably 99% ok :)